src/recipe.js

```diff
@@ -7,10 +9,17 @@ class Recipe {
     this.ingredients = recipe.ingredients;
   }
   calculateIngredientsCost() {
-    // return this.ingredients.map(i => {
-    //   ingredientData.find(ingredient => ingredient === i);
-    // });
+    let totalInCents = this.ingredients.reduce((total, item) => {
+      let itemPrice = ingredientsDB.default.find(ingredient => ingredient.id === item.id).estimatedCostInCents;
+      return total += itemPrice * item.quantity.amount;
+    },0);
+    let dollars = totalInCents / 100;
+
+    return dollars = dollars.toLocaleString("en-US", {style:"currency", currency:"USD"});
```

Comment:
I though a reduce was better so that we did not have to work with two separate arrays.

Reply:
Does line 19 actually run? Seems like you're reassigning a value inside of a return, which I didn't think was allowed

Reply:
That line is being executed outside of the reduce. It's just converting the output of the reduce function to a dollar amount.